src/main/clojure/osgi/deps.clj

```diff
@@ -0,0 +1,390 @@
+       :description   (.getDescription f)
+       :details       (.getDetails f)
+       :version       (.getVersion f)
+       :resolver      (.getResolver f)
+       :install       (.getInstall f)
+       :isHidden      (.isHidden f)
+       :dependencies  (map (fn [d] {
+                                    :name         (.getName d)
+                                    :version      (.getVersion d)
+                                    :isPrereq     (.isPrerequisite d)
+                                    :isDependency (.isDependency d)})
+                           (.getDependencies f))
+       :bundles       (map (fn [b] {
+                                    :location         (.getLocation b)
+                                    :originalLocation (.getOriginalLocation b)
+                                    :startLevel       (.getStartLevel b)
+                                    :isStart          (.isStart b)
+                                    :isDependency     (.isDependency b)})
+                           (.getBundles f))
+       :startLevel    (.getStartLevel f)
+       :namespace     (.getNamespace f)
+       :resourceRepos (.getResourceRepositories f)
+       :repoUrl       (.getRepositoryUrl f)})
+
+; ##################################################################
+
+(defn bundles [f]
+  (->> (osgi/bundle-list)
+       (map osgi/bundle->map)
+       (filter f)))
+
+(defn features []
+  (->> (osgi/list-features)
+       (map feature->map)))
+
+(comment
+  (->> (bundles (fn [x] true))
+       (filter #(.contains (:name %) "directorymonitor"))
+       (first)))
+
+(comment
+  (->> (features)
+       (filter #(.contains (:name %) "directorymonitor"))
+       (first)))
+
+; ##################################################################
+
+(defn bundles-partitioned
+  "Example function that partitions bundles. Takes a
+  coll of bundle defs and returns a coll of bundle def colls."
+  [bundles]
+  (let [cushion 1
+        id-set (set (map :id bundles))]
+    (->> bundles
+         (partition-by
+           #(and
+              (contains? id-set (+ (:id %) cushion))
+              (contains? id-set (- (:id %) cushion)))))))
+(comment
+  (map :name (bundles (fn [x] true)))
+  (->> (bundles (fn [x] true))
+       (bundles-partitioned)
+       (map #(map :name %)))
+  #_ ())
+
+
+(defn get-cdm-bundle
+  "Fetches the Content Directory Monitor bundle. Useful
+  for examples and testing."
+  []
+  (->> (bundles (fn [x] true))
+       (filter #(.contains (:name %) "directorymonitor"))
+       (first)))
+
+
+(defn select-all
+  "Predicate that always returns true, effectively filters
+  nothing when a selection is needed. Works for any arg."
+  [any]
+  true)
+(comment (select-all []))
+
+
+(defn selectf-bundles-built-by
+  "Predicate factory for bundle defs that eval true if a match is found
+  against the name of the user who built the bundle."
+  [user]
+  (fn [b] (->> b (:headers) (:built-by) (= user))))
```

Comment:
✏️ Really didn't want to use the term "factory" in Clojure code 🤒 but I'm looking for a good naming convention that tells the caller "I'm a function that provides a predicate; invoke me when you pass me". 

Maybe `select-` vs `selector-` or `selectable-` but I'm not sure, so I went with `selectf` ... i.e. the "select" function. Not a fan.